src/main/java/htsjdk/samtools/SamInputResource.java

```diff
@@ -91,7 +92,12 @@ public static SamInputResource of(final File file) {
 
     /** Creates a {@link SamInputResource} reading from the provided resource, with no index. */
     public static SamInputResource of(final Path path) {
-        return new SamInputResource(new PathInputResource(path));
+
+        if (Files.isRegularFile(path) &&  Files.exists(path)) {
```

Comment:
@yfarjoun Could you open a separate PR to remove the `Files.exists()` check here? This operation is expensive over certain NIO filesystems (eg., takes on the order of seconds over GCS), so forcing it at reader creation time could potentially add hours to the wall-clock time of a task that needed to access many bams over GCS. 

It's also not particularly logical to go to the else clause here if the file doesn't exist -- if it's a non-existent GCS file, `PathInputResource` is better equipped to produce a more relevant error message.